### ISSUE TITLE
[Feature] Add BatchNorm layer

### ISSUE THREAD

**Issue by zaleslaw** [JetBrains/KotlinDL#34]
Keras framework has a BatchNorm layer, and it's a significant and widely used layer in modern CNN architectures.

Unfortunately, the TF Java API 1.15 and TF Java API 2.x wraps outdated BatchNorm operand; alternative FusedBatchNorm is a real operand at all and could not be used as part of the trained graph.

The best option here creates our own BatchNorm operand based on another low-level TF operation with the ability to participate in backward and forward calculations.

**Comment by zaleslaw** [comment]
Unfortunately, FusedBatchNorm doesn't have gradients support in TF C/C++ API, the basic implementation for working in inference mode is implemented and merged.
